modules/voltage-divider.stanza

```diff
@@ -134,13 +138,18 @@ public defn check-output-voltage-range (v-in: [Double, Double, Double],
                                                               min-lo-dr, max-lo-dr, min-hi-dr, max-hi-dr)
   v-out[0] <= vo-min and vo-max <= v-out[2]
 
-defn query-resistors (resistance: Double, tol: Double) -> Tuple<Resistor> :
-  Resistors(["resistance" => resistance,
-             "tolerance" => tol,
-             "minimum_quantity" => 1,
-             "mounting" => PREFERRED-MOUNTING,
-             "case" => get-valid-pkg-list(),
-             "min-stock" => 5 * DESIGN-QUANTITY], ["tcr"])
+defn query-resistors (resistance: Double, tol: Double, params:Tuple<KeyValue<String, ?>>) -> Tuple<Resistor> :
+  val defaults = [
+    "resistance" => resistance,
+    "tolerance" => tol,
+    "minimum_quantity" => 1,
+    "mounting" => PREFERRED-MOUNTING,
+    "case" => get-valid-pkg-list(),
+    "min-stock" => 5 * DESIGN-QUANTITY
+  ]
+  val params* = to-tuple $ 
+    cat(defaults, filter({not contains?(defaults, _)}, params))
```

Comment:
I tested this, contains? is comparing KeyValue objects instead of keys. Plus I think you want params to override defaults not the opposite. You can use `to-tuple $ to-hashtable<String, ?> $ cat(defaults, params)`